modules/InboundEmail/InboundEmail.php

```diff
@@ -5144,15 +5140,37 @@ public function returnNonImportedEmail($msgNo, $uid)
             $structure = imap_fetchstructure($this->conn, $uid, FT_UID); // map of email
             $email->name = $this->handleMimeHeaderDecode($header[0]->subject);
             $email->type = 'inbound';
-            if (!empty($unixHeaderDate)) {
-                $email->date_sent = $timedate->asUser($unixHeaderDate);
-                list($email->date_start, $email->time_start) = $timedate->split_date_time($email->date_sent);
 
-            } else {
-                $email->date_start = $email->time_start = $email->date_sent = "";
-            }
 
+            if(empty($email->date_entered)) {
+
+                // find if string has (UTC) in timezone
+                $pattern = '/([(])([A-Z])+([)])/i';
+                $timezoneTextFound = preg_match($pattern,  $header[0]->date);
+
+                $dateTimeFormat = 'D, d M Y H:i:s O *';
+                if($timezoneTextFound === false || $timezoneTextFound === 0) {
+                    $dateTimeFormat = 'D, d M Y H:i:s O';
+                }
 
+                $dateTime = DateTime::createFromFormat(
+                    $dateTimeFormat,
+                    $header[0]->date
+                );
+
```

Comment:
php.net:

> 'DateTime::createFromFormat()' Returns a new DateTime instance or FALSE on failure.

so please, handle the error, for e.g:
`if($dateTime === false) { /* do something for handling the error date format... */ }`